Essence of Linear Algebra: Publish Cramer's Rule (#410)
3b1b/3Blue1Brown.com: public/content/lessons/2019/cramers-rule/index.mdx

source: _2018/cramer.py
credits:
- Lesson by Grant Sanderson
- Text adaptation by Kurt Bruns
- Text adaptation by James Schloss
---

> Jerry: Ah, you're crazy!
>
> Kramer: Am I? Or am I so sane that you just blew your mind?
>
> Jerry: It's impossible!
>
> Kramer: Is it?! Or is it so possible your head is spinning like a top?

> In a previous chapter, we talked about linear systems of equations, and sort of brushed aside the discussion of actually computing solutions to these systems. While it's true that number-crunching is something we typically leave to the computers, digging into some of these computational methods is a good litmus test for whether or not you actually understand what's going on, since this is really where the rubber meets the road.

Reviewer: ```suggestion
In a previous chapter, we talked about linear systems of equations, and sort of brushed aside the discussion of actually computing solutions to these systems. While it's true that number-crunching is something we typically leave to the computers, digging into some of these computational methods is a good test for whether or not you understand what's going on since this is where the rubber meets the road.
```

<Figure
    image="./figures/introduction/ComputingSolutions.light.svg" 
    width="480px"
/>

Here we want to describe the geometry behind a certain method for computing solutions to these systems, known as Cramer's rule. The relevant background needed here is an understanding of <LessonLink id="determinant">determinants</LessonLink>, <LessonLink id="dot-products">dot products</LessonLink>, and of <LessonLink id="inverse-matrices">linear systems of equations</LessonLink>, so be sure to read the relevant chapters on those topics if you're unfamiliar or rusty.

> But first! We should say up front that Cramer's rule is not the best way for computing solutions to linear systems of equations. Gaussian elimination, for example, will always be faster. So why learn it?

Reviewer: ```suggestion
But first! We should say up front that Cramer's rule is not the best way to compute solutions to linear systems of equations. Gaussian elimination, for example, will generally be faster, especially for larger matrices. So why learn it?
```

The setup here will be some linear system of equations, say with two unknowns, $x$ and $y$, and two equations. In principle, everything we're talking about will work systems with a larger number of unknowns, and the same number of equations. But for simplicity, a smaller example is nicer to hold in our heads.

<Figure
    image="./figures/introduction/SystemOfEquations.light.svg" 
    width="480px"
/>

As we talked about in a previous chapter, you can think of this setup geometrically as a certain known matrix transforming an unknown vector, $\left[\begin{array}{c} x \\ y \end{array}\right]$, where you know what the output is going to be, in this case $\left[\begin{array}{c} -4 \\ -2 \end{array}\right]$. Remember, the columns of this matrix tell you how the matrix acts as a transform, each one telling you where the basis vectors of the input space land.

<Figure
    image="./figures/introduction/GeometricSetup.light.svg" 
/>

> This is a sort of puzzle, what input $\left[\begin{array}{c} x \\ y \end{array}\right]$, is going to give you this output $\left[\begin{array}{c} -4 \\ -2 \end{array}\right]$?

Reviewer: ```suggestion
This is a puzzle. What input $\left[\begin{array}{c} x \\ y \end{array}\right]$, is going to give you this output $\left[\begin{array}{c} -4 \\ -2 \end{array}\right]$?
```
